server/actions/MongoDriver.ts

```diff
@@ -13,10 +21,19 @@ const ENTITY_NOT_FOUND_MESSAGE = 'Entity does not exist'
  */
 export async function getEntity<Schema extends Document>(
   dbSchema: Model<Schema>,
-  id: string
-) {
+  id: string,
+  aggregate?: PipelineStage[]
+): Promise<Schema & { _id: ObjectId }> {
   await mongoDb()
 
+  // if aggregate is defined, use that instead of simple find
+  if (!!aggregate) {
+    const objectId = new Types.ObjectId(id)
+    aggregate.push({ $match: { _id: objectId } })
+    const response = await dbSchema.aggregate(aggregate)
```

Comment:
need to return this response, and should only deal with `response[0]`. since this is only getting a single entity by id. Since we added the return types to the driver funcs, this also conflicts with that return type